azure/scope/machine.go

```diff
@@ -371,6 +382,51 @@ func (m *MachineScope) ProviderID() string {
 	return parsed.String()
 }
 
+// AvailabilitySet returns the availability set for this machine if available.
+func (m *MachineScope) AvailabilitySetSpec() (azure.ResourceSpecGetter, error) {
+	if !m.AvailabilitySetEnabled() {
+		// Availability set not enabled, not an error
+		return nil, nil
```

Comment:
The original function `AvailabilitySet() (string, bool)` returns the name and true/false whether the availabilty set is available for the machine. Since it's not an error, this return `nil` to keep the same behavior pattern. Is there a better way to handle this?

Reply:
why duplicate the code from the AvailabilitySet() function below and not just use it directly? ie

```
availabilitySetName, ok := s.Scope.AvailabilitySet()
if !ok {
	return nil
}

spec := &availabilitysets.AvailabilitySetSpec{
		Name: availabilitySetName,
		ResourceGroup:  m.ResourceGroup(),
		ClusterName:    m.ClusterName(),
		Location:       m.Location(),
		AdditionalTags: m.AdditionalTags(),
	}

...
```